howso/ablation.amlg

```diff
@@ -139,6 +139,77 @@
 		(call !CreateCaseWeights (assoc feature_name weight_feature) )
 	)
 
+	;reduce a model by removing cases which have an influence weight entropy that falls above
+	; a threshold
+	;
+	;parameters:
+	; context_features: a list of context features to use when computing influence weight entropies
+	; distribute_weight_feature: optional, default '.case_weight'. name of feature whose values to use as case weights
+	; influence_weight_entropy_threshold: optional, default 0.6. quartile to use as the cutoff threshold for
+	;influence weight entropy when deciding which cases to remove.
+	#reduce_model
+	(declare
+		(assoc
+			context_features (list)
+			distribute_weight_feature (null)
+			influence_weight_entropy_threshold (null)
+
+			;internal
+			max_influence_weight_entropy_to_keep .infinity
+		)
+
+		(assign (assoc
+			distribute_weight_feature
+				(if
+					(= (null) distribute_weight_feature)
+					!autoAblationWeightFeature
+					distribute_weight_feature
+				)
+			influence_weight_entropy_threshold
+				(if
+					(= (null) influence_weight_entropy_threshold)
+					!autoAblationInfluenceWeightEntropyThreshold
+					influence_weight_entropy_threshold
+				)
+		))
+
+		(if
+			(not !hasPopulatedCaseWeight)
+			(call !InitializeAutoAblation)
+		)
+		(if
+			(not !hasInfluenceWeightEntropies)
+			(call !ComputeAndStoreInfluenceWeightEntropies)
+		)
+
+		(assign (assoc
+			max_influence_weight_entropy_to_keep
+				(compute_on_contained_entities (list
+					(query_exists !internalLabelInfluenceWeightEntropy)
+					(query_quantile
+						!internalLabelInfluenceWeightEntropy
+						influence_weight_entropy_threshold
+						distribute_weight_feature
+					)
+				))
+		))
+
+		(call_entity trainee "!RemoveCases" (assoc
+			cases
+				(call !GetCasesByCondition (assoc
+					condition
+						(associate
```

Comment:
just to make sure this is intentional - you are explicitly creating a temporary local variable with the same name to shadow the model parameter just for the duration of this call?  If this is intentional, please add a comment to explain why.  Shadowing variables like that isn't good practice IMO.

Reply:
I don't follow. I'm using `associate` so that `!internalLabelInfluenceWeightEntropy` gets evaluated since the condition `assoc` needs to be  a feature name -> a list of two values representing the range of cases that satisfy the condition. So it ends up being

`(assoc influence_weight_entropy (list 1.045 .infinity)` or something similar.